### PR TITLE
Stow task outputting final json with FCN

### PR DESCRIPTION
**Pull request by knorth55**
*waiting for upstream*

## What is new
- pick largest object segmented in FCN
- save stow task item location in json ( #2178)
  - add `json_saver.py` and `UpdateJSON` `SaveJSON` services

## Minor changes
- update stow task env config
- remove unused params in `setup_stow.launch`
- correct indent in `stow-interface.l`

## Requirement
https://github.com/jsk-ros-pkg/jsk_recognition/pull/2133 and https://github.com/jsk-ros-pkg/jsk_recognition/pull/2134

## How to use

```
roslaunch jsk_arc2017_baxter baxter.launch pick:=false
roslaunch jsk_arc2017_baxter setup_for_stow.launch
roslaunch jsk_arc2017_baxter stow.launch json_dir:=~~~~~
```

## Videos
https://drive.google.com/open?id=0B5DV6gwLHtyJN01oeEFBNG5WNFE
https://drive.google.com/open?id=0B5DV6gwLHtyJRk5JdVh0OUQ4V0U

### CODE REVIEW

File: jsk_arc2017_common/srv/UpdateJSON.srv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+string item
+string bin
```
Comment (wkentaro) [inline jsk_arc2017_common/srv/UpdateJSON.srv]: Please add `from_location` and `to_location`.

File: jsk_arc2017_common/srv/SaveJSON.srv [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
+bool saved
```
Comment (wkentaro) [inline jsk_arc2017_common/srv/SaveJSON.srv]: Please use `std_srvs/Trigger`.

File: jsk_arc2017_common/node_scripts/json_saver.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        if json_dir is None:
+            rospy.logerr('must set json dir path to ~json_dir')
+            return
+        if output_dir is None:
+            rospy.logerr('must set output dir path to ~output_dir')
+            return
+        if not osp.exists(output_dir):
+            os.mkdir(output_dir)
+        location_path = osp.join(json_dir, 'item_location_file.json')
+        self.output_json_path = osp.join(output_dir, 'item_location_file.json')
```
Comment (wkentaro) [inline jsk_arc2017_common/node_scripts/json_saver.py]: Why not create subdirectory with timestamp?

```
osp.join(output_dir, '%s' % rospy.Time.now().to_nsec(), 'item_location_file.json')
```

File: jsk_arc2017_baxter/euslisp/lib/arc-interface.l [hunk 4]
```diff
@@ -212,6 +214,22 @@
...
+                                        :timeout 10000
+                                        :after-stamp stamp))
+      (dolist (box (send box-msg :boxes))
+        (let (volume)
+          (setq volume (* (send box :dimensions :x)
+                          (send box :dimensions :y)
+                          (send box :dimensions :z)))
+          (if (> bbox-volume-threshold- volume)
+            (return-from :get-largest-object-index
+                         (+ (send box :label) 1)))))))
```
Comment (pazeshun) [inline jsk_arc2017_baxter/euslisp/lib/arc-interface.l]: Why do you use `return-from` ?

Comment (knorth55) [inline jsk_arc2017_baxter/euslisp/lib/arc-interface.l]: This is same as break insolist

Comment (pazeshun) [inline jsk_arc2017_baxter/euslisp/lib/arc-interface.l]: I understand.

**Comment by knorth55** [comment]
I updated FCN recognition pipeline and add [WIP] label.
I will test this on real robot.

**Comment by knorth55** [comment]
I revert recognition pipeline change and remove WIP label

**Comment by knorth55** [comment]
remove wait4upstream label

**Pull request merged**